### PR TITLE
CO-2403_Prevent_Enrollment_flow_for_authenticated_user

### PR DESCRIPTION
**Pull request by ioigoume**
Duplicate Account Check Enroller Plugin

### CODE REVIEW

File: app/AvailablePlugin/DuplicateAccountEnroller/Config/Schema/schema.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  @since         COmanage Registry v4.3.0
+  @license       Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+-->
+<!--
+  While this schema mostly works with database prefixes, foreign key constraints
+  must be specified in raw SQL, which needs the prefixed table name.
+-->
+
+<schema version="0.3">
+  <table name="duplicate_account_enrollers">
```
Comment (boshrin) [inline app/AvailablePlugin/DuplicateAccountEnroller/Config/Schema/schema.xml:33]: "account" doesn't mean anything in a COmanage context, so we should avoid using that term. The plugin should be called something like "DuplicateCheckEnroller"

File: app/AvailablePlugin/DuplicateAccountEnroller/Config/Schema/schema.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    <field name="id" type="I">
+      <key />
+      <autoincrement />
+    </field>
+    <field name="co_enrollment_flow_wedge_id" type="I">
+      <constraint>REFERENCES cm_co_enrollment_flow_wedges(id)</constraint>
+    </field>
+    <field name="redirect_url" type="C" />
+    <field name="type" type="C" size="32" />
```
Comment (boshrin) [inline app/AvailablePlugin/DuplicateAccountEnroller/Config/Schema/schema.xml:42]: `type` should have a more descriptive name, eg `identifier_type`.

File: app/AvailablePlugin/DuplicateAccountEnroller/Model/DuplicateAccountEnroller.php [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+   * Check for CO Person Identifier duplicates.
+   *
+   * @since  COmanage Registry v4.3.0
+   * @param  integer $coId                    CO ID
+   * @param  string  $env_remote_user         Identifier from Env
+   * @param  string  $ident_type              Identifier Type
+   * @return Array
+   */
+
+  public function findCoPersonDuplicate($coId, $env_remote_user, $ident_type) {
```
Comment (boshrin) [inline app/AvailablePlugin/DuplicateAccountEnroller/Model/DuplicateAccountEnroller.php:106]: findXXX prefixes can conflict with Cake's magic finding, these functions should probably have a different prefix (eg: `searchCoPersonDuplicates`).

File: app/AvailablePlugin/DuplicateAccountEnroller/View/DuplicateAccountEnrollers/fields.inc [hunk 4]
```diff
@@ -0,0 +1,175 @@
+<!--
+/**
+ * COmanage Registry LIGO MOU Transfer Enrollers Plugin Fields
```
Comment (boshrin) [inline app/AvailablePlugin/DuplicateAccountEnroller/View/DuplicateAccountEnrollers/fields.inc:3]: copy/paste error

File: app/AvailablePlugin/DuplicateAccountEnroller/View/DuplicateAccountEnrollers/fields.inc [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ * @link          http://www.internet2.edu/comanage COmanage Project
+ * @package       registry-plugin
+ * @since         COmanage Registry v4.1.0
```
Comment (boshrin) [inline app/AvailablePlugin/DuplicateAccountEnroller/View/DuplicateAccountEnrollers/fields.inc:24]: copy/paste error

**Review by boshrin — CHANGES_REQUESTED** [review]

File: app/AvailablePlugin/DuplicateAccountEnroller/View/DuplicateAccountEnrollers/fields.inc (hunk 4 above)
Comment (ioigoume) [reply app/AvailablePlugin/DuplicateAccountEnroller/View/DuplicateAccountEnrollers/fields.inc]: Done

File: app/AvailablePlugin/DuplicateAccountEnroller/Model/DuplicateAccountEnroller.php (hunk 3 above)
Comment (ioigoume) [reply app/AvailablePlugin/DuplicateAccountEnroller/Model/DuplicateAccountEnroller.php]: Done

File: app/AvailablePlugin/DuplicateAccountEnroller/Config/Schema/schema.xml (hunk 2 above)
Comment (ioigoume) [reply app/AvailablePlugin/DuplicateAccountEnroller/Config/Schema/schema.xml]: Done

File: app/AvailablePlugin/DuplicateAccountEnroller/Config/Schema/schema.xml (hunk 1 above)
Comment (ioigoume) [reply app/AvailablePlugin/DuplicateAccountEnroller/Config/Schema/schema.xml]: Done

**Pull request merged**